feat: mypage-setting: 회원탈퇴

MillPRE opened the pull request:
+ 회원탈퇴 기존 디자인과 다른 모달 변경 
+ 회원탈퇴 api 연동 및 탈퇴 성공시 렌더링 페이지로 라우팅

MillPRE committed:
feat: mypage-setting: 회원탈퇴

su-lim commented on src/pages/mypage/SettingPage.tsx:
```diff
@@ -4,6 +4,9 @@ import { TopBar } from '../../components/common/Navigation';
 import axios from 'axios';
+import CustomModal from "../../components/common/CustomModal";
+import Button from "../../components/Button/Button";
+import {CopyToClipboard} from "react-copy-to-clipboard";
```
쓰지 않는 AlertModal, CopyToClipboard를 import하는 코드는 삭제하면 좋을 것 같습니당!

su-lim commented on src/pages/mypage/SettingPage.tsx:
```diff
@@ -4,6 +4,9 @@ import { TopBar } from '../../components/common/Navigation';
+import {CopyToClipboard} from "react-copy-to-clipboard";
 
 function ChangePasswordModal() {
   return <div className={'password-change-modal-container'}>qwe</div>;
```
여기서 qwe는 어떤 의미인가요?

su-lim commented on src/pages/mypage/SettingPage.tsx:
```diff
@@ -15,27 +18,42 @@ const SettingPage = () => {
     password: false,
   });
   const [quit, setQuit] = useState(false); // 탈퇴 state
+  const [ userNickName, setUserNickName] = useState(''); // user nick name
```
userNickName을 선언했는데 코드 내에서 쓰이지 않고 있어요! userNickName을 써야하는데 놓치신건지, 이 state가 없어도 문제가 없는지(그러면 관련 코드 삭제) 다시 확인해보시면 좋을 것 같습니다 :)

su-lim commented on src/pages/mypage/SettingPage.tsx:
```diff
@@ -15,27 +18,42 @@ const SettingPage = () => {
+        }
+      })
+      .catch((err) => {
+        alert('탈퇴에 실패하였습니다.');
```
서비스에서 사용자와 계속 BasicModal, CustomModal을 통해 소통해왔으니, 이번에도 alert창 보다는 사전에 구현한 Modal을 쓰면 좋을 것 같습니다! 참고하실 코드입니당

```
 <BasicModal
  open={state}
  setOpen={setState}
  title="모달제목"
  ok="모달 확인 버튼 텍스트"
  okAction={확인 버튼 클릭시 동작 함수} // e.g. ()=>setState(false), ()=>window.location~
></BasicModal>
```

su-lim commented on src/pages/mypage/SettingPage.tsx:
```diff
@@ -104,9 +125,7 @@ const SettingPage = () => {
         </h3>
         <h3
-          onClick={() => {
-            setAlert({ quit: true, password: false });
```
이 이벤트를 통해서 저희는 (1) password 창을 띄우고 싶은 것이고, (2) Alert 선언 시 속성에 내용이 바뀔 경우 이 코드도 찾아서 바꾸어야 할 것 같아요!
그래서 저같은 경우는, 바꾸려는 속성과(password: true)와 현재의 속성을 모두 적는 것보다는
현재 상태에서 `{...Alert}`, 바꾸려는 부분만 `, password: true` 적는 편입니다!

이렇게 작성하시면 코드를 읽을 때도 어떤 동작을 의도했는지 파악하기 쉽고, 나중에 다시 수정할 일이 적어져서 좋을 것 같아요!

MillPRE replied:
오오.. 감사함다.. spread operation 사용하는것이 익숙하지 않아서 실제 코딩할때는 많이 사용하지 않았었는데, 이런식으로 활용하면 코드 가독성도 높아져서 더 좋을 것 같네용 !

su-lim reviewed (commented):
수고 많으셨습니다 :) 리뷰 확인 해보시고 이해가 안되는 말이나 다른 의견이 있으시면 알려주세요~

+) 

1. Alert State에서 결과적으로 quit는 쓰이지 않는데 객체 형태의 Alert가 아닌 String 형태의 password(가칭) State만 사용하도록 바꿀 수 있을 것 같습니다!

2. 비밀번호 변경 창에서 뒤로가기 시 직전이었던 설정 창이 아닌 마이페이지로 가게되어서 수정해야할 것 같습니다. 제 생각에 설정창 안에서 state로 비밀번호 변경 창을 보여주는 것이 아닌 비밀번호 변경 창을 새로운 페이지로 분리하여 만들어도 될 것 같습니다! (e.g. setting-password/ 등)

MillPRE committed:
fix: api(followers) request error fix

+ myInfo.nickname 값을 인자로하는 api에서 myInfo 값이 아직 set되지 않은 채로 요청이감
+ myInfo.nickname값이 set되야만 요청이 가도록 수정

MillPRE committed:
chore: 비밀번호 변경 페이지화

+ 비밀번호 변경하는 부분을 페이지로 변경하여 비밀번호 변경 페이지에서 뒤로가기 버튼 클릭시 /mypage/setting으로 이동하게 변경

MillPRE committed:
chore: 사용하지 않는 프로필 셋팅 페이지 파일 삭제
